obs2ioda-v2/src/ncio_mod.f90

```diff
@@ -20,107 +22,134 @@ module ncio_mod
 
 contains
 
-subroutine write_obs (filedate, write_opt, outdir, itim)
+function get_dim_name(dimid,nchans_nvars_flag) result(dim_name)
+   integer(i_kind),intent(in) :: dimid
+   logical,intent(in) :: nchans_nvars_flag
+   character(nstring) :: dim_name
+   character(nstring) :: tmp
+   tmp = name_ncdim(dimid)
+   if ( nchans_nvars_flag) then
+      if ( trim(tmp) == 'nvars') then
+         dim_name = 'nchans'
+      else
+         dim_name = tmp
+      end if
+   else
+      dim_name = tmp
+   end if
+end function get_dim_name
```

Comment:
It looks like the implementation of this function has reverted to its earlier form.